bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -56,6 +56,16 @@
  */
 public class BigtableDataGrpcClient implements BigtableDataClient {
 
+  private class ImmutableListFunction<T> implements Function<List<T>, ImmutableList<T>> {
```

Comment:
This is generic usage, maybe name it that?
ImmutableListCopier? ImmutableListCopierFunction?

List<T> items instead of List<T> sampleRowKeysResponses